addons/l10n_it_edi/tests/test_edi_import.py

```diff
@@ -71,6 +72,28 @@ def test_receive_signed_vendor_bill(self):
             }],
         }])
 
+    def test_receive_bill_sequence(self):
+        """ Ensure that the received bill gets assigned the right sequence. """
+        def mock_commit(self):
+            pass
+
+        filename = 'IT01234567890_FPR02.xml'
+        with (patch.object(self.proxy_user.__class__, '_decrypt_data', return_value=self.fake_test_content),
+              patch.object(sql_db.Cursor, "commit", mock_commit),
+              tools.mute_logger("odoo.addons.l10n_it_edi.models.account_move"),
+              freeze_time('2019-01-01')):
+            self.env['account.move']._l10n_it_edi_process_downloads({
+                '999999999': {
+                    'filename': filename,
+                    'file': self.fake_test_content,
+                    'key': str(uuid.uuid4()),
+                }},
+                self.proxy_user,
+            )
+
+        invoices = self.env['account.move'].search([('payment_reference', '=', 'TWICE_TEST')])
+        self.assertEqual(invoices.name, 'BILL/2019/01/0001')
```

Comment:
```suggestion
        invoices = self.env['account.move'].with_company(self.company).search([('name', '=', 'BILL/2019/01/0001')])
        self.assertEqual(len(invoices), 1)
```

Please do not search by `payment_reference`, but directly with the object of your testing. Tests are 60% avoiding regressions, 40% documentation of behaviours, it's important to be clear.

Reply:
Now you're searching for the attachment... ? Why not by name?

Reply:
I wanted to make sure that the bill I'm testing is actually the one that was generated, in case there's already a bill named `BILL/2019/01/0001`. This should not happen, but I wanted to make sure.

Reply:
Then you can search for that name before and after. Don't test two things with one test.

Reply:
Done, thanks !